app/javascript/components/patient/follow_up_flag/FollowUpFlag.js

```diff
@@ -0,0 +1,272 @@
+          })
+          .catch(error => {
+            reportError('Failed to bulk update Follow-up Flag.');
+            console.log(error);
+          });
+      } else {
+        axios
+          .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/follow_up_flag', {
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.apply_to_household,
+            apply_to_household_ids: this.state.apply_to_household_ids,
+          })
+          .then(() => {
+            // Reload the page to see the flag on the monitoree's page
+            location.reload();
+          })
+          .catch(error => {
+            reportError(error?.response?.data?.error ? error.response.data.error : error, false);
+            console.log(error);
+          });
+      }
+    });
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <Modal.Body>
+          <Form.Group style={{ display: 'inline-flex' }}>
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              className="pr-5"
+              id="set_flag_for_follow_up"
+              label="Set Follow-up Flag"
+              onChange={this.handleChange}
+              checked={!this.state.clear_flag}
+            />
+            <Form.Check
+              type="radio"
+              name="flag_for_follow_up_option"
+              id="clear_flag_for_follow_up"
+              label="Clear Follow-up Flag"
+              disabled={this.state.clear_flag_disabled}
+              onChange={this.handleChange}
+              checked={this.state.clear_flag}
+            />
+          </Form.Group>
+          {!this.state.clear_flag && (
+            <Form.Group>
+              <Form.Label>
+                Please select a reason for being flagged for follow-up. If a monitoree is already flagged, this reason will replace any previously selected
+                reason.
+              </Form.Label>
+              <Form.Control
+                as="select"
+                size="lg"
+                className="form-square mb-3"
+                id="follow_up_reason"
+                value={this.state.follow_up_reason}
+                onChange={this.handleChange}>
+                <option></option>
+                <option>Deceased</option>
+                <option>Duplicate</option>
+                <option>High-Risk</option>
+                <option>Hospitalized</option>
+                <option>In Need of Follow-up</option>
+                <option>Lost to Follow-up</option>
+                <option>Needs Interpretation</option>
+                <option>Quality Assurance</option>
+                <option>Other</option>
+              </Form.Control>
+              <Form.Group controlId="follow_up_note">
```

Comment:
Is there a reason this `Form.Group` is contained in another `Form.Group`?  I'd move it out and wrap the two `Form.Group`s in the condition in a `React.Fragment`

Reply:
I took your suggestion, it should be good now!